backend/src/services/users.py

```diff
@@ -37,3 +41,38 @@ async def update_user(self, user_id: UUID, user: UserUpdate) -> User | dict[str,
     async def delete_user(self, user_id: UUID) -> dict[str, str]:
         obj = await self.user_manager.delete(obj_id=user_id)
         return {"message": "User was successfully deleted"} if isinstance(obj, User) else obj
+
+    async def check_if_user_exists(self, email: str) -> Never | None:
+        if await self.user_manager.get_by_email(filters={"email": email}):
+            LOG.error(f"User with {email=} already exists exception")
+            raise UserAlreadyExistsException()
+
+    async def register_user(self, user: UserRegister) -> User | dict[str, str]:
+        await self.check_if_user_exists(user.email)
+
+        hashed_password = make_password_hash(user.password)
+        user = user.model_copy(update={"password": hashed_password})
+        return await self.create_user(user=user)
+
+    async def authenticate_user(self, email: str, password: str) -> User | bool:
+        user = await self.user_manager.get_by_email(filters={"email": email})
+        if not user or not verify_password(password, user.password):
+            return False
```

Comment:
Why can't you just use?
`user = await self.user_manager.get_by(filters={"email": email})`

Reply:
Changed to `get_by`